File: app/src/main/java/com/maximus/productivityappfinalproject/framework/db/AppLimitDao.java
package com.maximus.productivityappfinalproject.framework.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.maximus.productivityappfinalproject.domain.model.AppUsageLimitModel;

import java.util.List;

@Dao
public interface AppLimitDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAppLimit(AppUsageLimitModel limitModel);

    @Query("SELECT * FROM app_limit_table")
    List<AppUsageLimitModel> getUsageLimitList();

    @Query("DELETE FROM app_limit_table WHERE packageName =:packageName")
    void deleteAppLimit(String packageName);
}

File: app/src/main/java/com/maximus/productivityappfinalproject/framework/db/AppLimitDao.kt
package com.maximus.productivityappfinalproject.framework.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.maximus.productivityappfinalproject.domain.model.AppUsageLimitModel
import io.reactivex.Flowable

@Dao
interface AppLimitDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAppLimit(limitApp: AppUsageLimitModel)

    @Query("SELECT * FROM app_limit_table")
    fun getUsageLimitList(): List<AppUsageLimitModel>

    @Query("DELETE FROM app_limit_table WHERE packageName =:packageName")
    fun deleteAppLimit(packageName: String)
}